src/app/tags/tags/tags.component.ts

```diff
@@ -71,7 +74,15 @@ export class TagsComponent implements OnInit {
       });
   }
 
-  editTag() {
-    console.log("editTag");
+  editTag(tag) {
+    const editTagDialog = this.dialog.open(TagDialogComponent, {
+      data: { action: "Edit", dataToEdit: tag, tags: this.tags },
+      minWidth: "50%"
+    });
+    editTagDialog.afterClosed().subscribe(res => {
+      if (res) {
+        this.tagsService.updateData(res).subscribe();
+      }
+    });
```

Comment:
rows 78 - 86 are almost duplicate. isn't is better to make one method for opening dialog with variables `action`, `tags` and `dataToEdit`?

Reply:
I don't think, that pass "action" in dialog and check this variable in result is better than two separate methods.

Reply:
Ok